contrib/test_mergeBenchmarkSets/test_mergeBenchmarkSets.py

```diff
@@ -0,0 +1,538 @@
+witness_xml_2 = ET.fromstring(mock_witness_2)  # noqa S314, the XML is trusted
+
+files = [
+    "../sv-benchmarks/c/array-examples/sanfoundry_24-1.yml",
+    "../sv-benchmarks/c/array-examples/data_structures_set_multi_proc_trivial_ground.yml",
+    "../sv-benchmarks/c/array-patterns/array28_pattern.yml",
+    "../sv-benchmarks/c/reducercommutativity/rangesum05.yml",
+    "../sv-benchmarks/c/array-fpi/indp4f.yml",
+]
+
+
+def mock_witness_sets():
+    witness_sets = {}
+    for witness in [witness_xml_1, witness_xml_2]:
+        for run in witness.findall("run"):
+            name = run.get("name")
+            witness_sets[name] = run
+    return [witness_sets]
+
+
+def mock_get_verification_result(name):
+    return results_xml.find("run[@name='{}']".format(name))
+
+
+def mock_get_witness(name):
+    witness = mock_witness_sets()[0].get(name)
+    if witness is None:
+        raise NotImplementedError(name)
+    return witness
+
+
+def element_trees_equal(et1, et2):
+    if len(et1) != len(et2) or et1.tag != et2.tag or et1.attrib != et2.attrib:
+        return False
+    return all(element_trees_equal(child1, child2) for child1, child2 in zip(et1, et2))
+
+
+class TestMergeBenchmarkSets(unittest.TestCase):
+    def test_only_elem(self):
+        new_results = mergeBenchmarkSets.xml_to_string(results_xml)
+        new_witness_1 = mergeBenchmarkSets.xml_to_string(witness_xml_1)
+        new_witness_2 = mergeBenchmarkSets.xml_to_string(witness_xml_2)
+        self.assertTrue(
+            element_trees_equal(
+                ET.fromstring(new_results), results_xml  # noqa S314, the XML is trusted
+            )
+        )
+        self.assertTrue(
+            element_trees_equal(
+                ET.fromstring(new_witness_1),  # noqa S314, the XML is trusted
+                witness_xml_1,
+            )
+        )
+        self.assertTrue(
+            element_trees_equal(
+                ET.fromstring(new_witness_2),  # noqa S314, the XML is trusted
+                witness_xml_2,
+            )
+        )
+
+    def test_set_doctype(self):
+        def assert_equal_ignore_space(fst, snd):
+            fst = [x.strip() for x in fst if x.strip()]
+            snd = [x.strip() for x in snd if x.strip()]
+            self.assertEqual(fst, snd)
+
+        qualified_name = "result"
+        public_id = "+//IDN sosy-lab.org//DTD BenchExec result 1.18//EN"
+        system_id = "https://www.sosy-lab.org/benchexec/result-1.18.dtd"
+        new_results = mergeBenchmarkSets.xml_to_string(
+            results_xml, qualified_name, public_id, system_id
+        )
+        new_witness_1 = mergeBenchmarkSets.xml_to_string(
+            witness_xml_1, qualified_name, public_id, system_id
+        )
+        new_witness_2 = mergeBenchmarkSets.xml_to_string(
+            witness_xml_2, qualified_name, public_id, system_id
+        )
+        self.assertTrue(element_trees_equal(results_xml, ET.fromstring(new_results)))
```

Comment:
Right now `flake8` complains about these calls. Using them in tests is no problem, so just silence `S314`. If you prefer, you can also do this for the whole file.

Reply:
Accidentaly committed this part, still looking into why one of the tests fails in CI (it doesn't on my machine). But yeah, I'll silence it.